src/components/sidebar/sidebar.test.tsx

```diff
@@ -0,0 +1,321 @@
+import React from "react";
+import {
+  render,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import {
+  testStyledSystemPadding,
+  testStyledSystemWidth,
+} from "../../__spec_helper__/__internal__/test-utils";
+import CarbonProvider from "../carbon-provider";
+import Form from "../form";
+
+import Sidebar, { SidebarProps } from ".";
+
+const AllTheProviders = ({ children }: { children: React.ReactNode }) => (
```

Comment:
**comment**: I really don't like the name `AllTheProviders` when there is only *one* provider involved here!

Further, it doesn't apepar to me that this particular wrapping is needed for any tests other than those that use the new validation pattern (which I'm not sure any of them do, having looked through all the tests in this file - none appear to use any input components at all). So it doesn't seem right to me that all the tests use this, if it's not necessary.

Finally, even if this is needed, and is renamed, since it's not particularly verbose I'd prefer to just use it explicitly in the `render` function: `render(<CarbonProvider validationRedesignOptIn><Sidebar {{...props}}/>{children}</Sidebar></CarbonProvider>`. I realise that is just a matter of opinion but I find it more readable, rather than having to go to the docs to figure out what `wrapper` is doing.

EDIT: I only just noticed this is also used in the `Dialog` tests, so perhaps it's better for consistency. I'm still quite unconvinced it's necessary or helpful though. Maybe there was a discussion I missed at the time of `Dialog` being converted to RTL?

Reply:
I'm happy to move the context provider into the render function of each test, especially since we are actively improving test reliability 👍🏼 

We do need the provider here though, mainly for supplying the [`TopModalContext` that `Sidebar` needs](https://github.com/Sage/carbon/blob/master/src/components/sidebar/sidebar.component.tsx#L141):

```tsx
// line 141
// hook consumes `TopModalContext` provided by `CarbonProvider`
const isTopModal = useModalAria(sidebarRef); 
```

Though the `validationRedesignOptIn` flag isn't necessarily required here.

Reply:
Ah yes, thanks - I should have realised that, since I was the one who added the `TopModalContext` into `CarbonProvider` 🤦‍♂️ 

In that case yes the provider is needed, but I agree to drop the `validationRedesignOptIn` and maybe (optionally, but I think it could be helpful to future readers) add a comment that this provider has been added specifically for the `TopModalContext`